librz/include/rz_util/ht_inc.h

```diff
@@ -188,7 +195,14 @@ RZ_API VALUE_TYPE Ht_(find)(RZ_NONNULL HtName_(Ht) *ht, const KEY_TYPE key, RZ_N
 // If the cb returns false, the iteration is stopped.
 // cb should not modify the hashtable.
 // NOTE: cb can delete the current element, but it should be avoided
-RZ_API void Ht_(foreach)(RZ_NONNULL HtName_(Ht) *ht, RZ_NONNULL HT_(ForeachCallback) cb, RZ_NULLABLE void *user);
+RZ_API void Ht_(foreach_cb)(RZ_NONNULL HtName_(Ht) *ht, RZ_NONNULL HT_(ForeachCallback) cb, RZ_NULLABLE void *user);
+RZ_API void Ht_(advance_iter)(RZ_NONNULL HtName_(Ht) *ht, RZ_NONNULL HT_(Iter) *it);
+
+#define ht_foreach(type, ht, iter) \
+	if (ht && ht_##type##_size(ht) > 0) \
+		for (iter.ti = 0, iter.bi = 0, iter.kv = NULL, ht_##type##_advance_iter(ht, &iter); iter.kv != NULL; ht_##type##_advance_iter(ht, &iter))
```

Comment:
IMO it would be preferable to have more standard iterator API with `first()/next()`
For example loop statement could look like:
```c
for (iter = Ht_(iter_first)(ht) /*As I see it is ok to return struct from API*/; iter.kv != NULL; Ht_(iter_next)(ht, &iter))
```
@XVilka @kazarmy @wargio @ret2libc your opinions?

Reply:
I like it! 👍

Reply:
Yes, makes sense. Check also:
- `librz/util/iterator.c`
- `librz/include/rz_util/rz_iterator.h`

Reply:
At this point we could also just add a `RzIterator *ht_xx_as_iter();`, right?
Otherwise, we just duplicate the API from iterators to hash tables.